Preferences > Decks: fix activation of rate change controls

ronso0 opened the pull request:
https://bugs.launchpad.net/mixxx/+bug/1931638
can't tell the reason for this change with git blame.
both the fine & coarse controls are available and working regardless of the pitchbend mode, so they should be configurable, too.

uklotzde reviewed (approved):
I also don't see any reason why these options should depend on the ramping behavior.

daschuer commented:
The Temporary spin boxes are only working when selecting "Abrupt jump" see:  
https://github.com/mixxxdj/mixxx/blob/9b9fbaafc404f8cfc822f64d356ed3c37e8778c1/src/engine/controls/ratecontrol.cpp#L516
In Smooth mode the ramping sensitivity slider is used. 

The preferences GUI is hard to understand here because it is not even obvious which GUI knob is affected by the setting. 

I think the original idea was to have either the slider or the temp spin boxes grayed out. 

A bug fix here would be to call setEnabled() for the widgets related to the Pitch bend behavior radio button.

daschuer reviewed (changes requested):
I would prefer to keep the graying out approach to have an idea how it fits together. 
On a long run we may dispose some of these config options and pick a default that works for all.

ronso0 committed:
Pref > Deck: fix enable/disable pitch bend widgets

ronso0 committed:
Pref > Deck: consolidate pitch bend setup

ronso0 committed:
Pref > Deck: rename confusing variables in update()

ronso0 committed:
RateControl: remove redundant rampDirection condition

ronso0 commented:
Alright, thanks fo the explanation @daschuer  (I probably shouldn't work on code in the sun anymore as trying to read the display sucks too much energy, and I get too lazy to do the digging myself... ; )

I restored and fixed the widget enabling/disabling. Checking the radio buttons was just done too early, before the connections were made. And checking the radio buttons was missing from update().

That's the first commit, the following are consolidating and cleaning up a bit, so next time this part is worked on it's more 'fun'.

Also, when looking into RateControl, I removed a redundant check there.

Even though this grow bigger than the actual bugfix, I hope it's easy to review and can slip in.
If not let me know and I'll isolate the first commit.

ronso0 commented:
> The preferences GUI is hard to understand here because it is not even obvious which GUI knob is affected by the setting.

yes, the layout is confusing in conjunction with the widgets being disabled.
IMO it would be easier to grasp when we break up that grid of spinboxes, put the temp spinboxes side by side below the slider, and the permanent spinboxes into a separate row with some margin, probably with its own sub-header.

uklotzde commented on src/engine/controls/ratecontrol.cpp:
```diff
@@ -535,26 +535,24 @@ void RateControl::processTempRate(const int bufferSamples) {
-                }
+            if (!m_bTempStarted) {
+                m_bTempStarted = true;
+                double latrate = ((double)bufferSamples / (double)m_pSampleRate->get());
```
If you touch the code please replace C-style with C++-style casts

ronso0 replied:
done.

ronso0 committed:
RateControl::processTempRate: use c++ style casts

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

ronso0 committed:
Pref > Deck: improve rate ramp controls arrangement

ronso0 commented:
Hopefully the relation of the controls is now a bit easier to understand:
![image](https://user-images.githubusercontent.com/5934199/122225451-5e370a80-ceb5-11eb-9d29-73159835502d.png)

A tooltip explaining the range of the sensitivity slider would improve that further,

daschuer commented on src/engine/controls/ratecontrol.cpp:
```diff
@@ -535,26 +535,25 @@ void RateControl::processTempRate(const int bufferSamples) {
-                }
+            if (!m_bTempStarted) {
+                m_bTempStarted = true;
+                double latrate = static_cast<double>(bufferSamples / m_pSampleRate->get());
```
This cast is redundant now, because the result from int/double is double already.

daschuer commented on src/engine/controls/ratecontrol.cpp:
```diff
@@ -535,26 +535,25 @@ void RateControl::processTempRate(const int bufferSamples) {
+            if (!m_bTempStarted) {
+                m_bTempStarted = true;
+                double latrate = static_cast<double>(bufferSamples / m_pSampleRate->get());
+                m_dRateTempRampChange = (latrate /
+                        (static_cast<double>(m_iRateRampSensitivity / 100.)));
```
```suggestion
                m_dRateTempRampChange = latrate / (m_iRateRampSensitivity / 100.0);
```

daschuer commented:
Works good, thank you. Just some nits about casting.

ronso0 commented:
done, @daschuer Changes squashed into https://github.com/mixxxdj/mixxx/pull/4002/commits/8ef7ff4872529c39c30781b8c9e2acd321557bed

daschuer commented:
Thank you :-)
